packages/dd-trace/src/config.js

```diff
@@ -627,7 +627,11 @@ class Config {
     target.otelMaxQueueSize = nonNegInt(OTEL_BSP_MAX_QUEUE_SIZE, 'OTEL_BSP_MAX_QUEUE_SIZE', false)
 
     const otelMetricsExporter = !OTEL_METRICS_EXPORTER || OTEL_METRICS_EXPORTER.toLowerCase() !== 'none'
-    this.#setBoolean(target, 'otelMetricsEnabled', DD_METRICS_OTEL_ENABLED && otelMetricsExporter)
+    this.#setBoolean(
+      target,
+      'otelMetricsEnabled',
+      DD_METRICS_OTEL_ENABLED && isTrue(DD_METRICS_OTEL_ENABLED) && otelMetricsExporter
```

Comment:
We use this pattern (`DD_METRICS_OTEL_ENABLED && isTrue(DD_METRICS_OTEL_ENABLED)`) in many places in our configs. It would be nice if we had a MaybeBool() helper that returned `undefined` if a config was not set. 

This change is beyond the scope of this PR